x-pack/plugins/ml/public/application/model_management/deployment_setup.tsx

```diff
@@ -251,28 +331,46 @@ interface StartDeploymentModalProps {
  * Modal window wrapper for {@link DeploymentSetup}
  */
 export const StartUpdateDeploymentModal: FC<StartDeploymentModalProps> = ({
-  modelId,
+  model,
   onConfigChange,
   onClose,
   startModelDeploymentDocUrl,
   initialParams,
 }) => {
+  const isUpdate = !!initialParams;
+
   const [config, setConfig] = useState<ThreadingParams>(
     initialParams ?? {
       numOfAllocations: 1,
       threadsPerAllocations: 1,
       priority: isCloudTrial() ? 'low' : 'normal',
+      deploymentId: model.model_id,
     }
   );
 
-  const isUpdate = initialParams !== undefined;
+  const deploymentIdValidator = useMemo(() => {
```

Comment:
Currently we only check on the client if the deployment ID is unique for this model and not across all models. Is it possible, or worthwhile, checking across all models? Note the backend does return an error on start if the ID is being used by a different model:

<img width="268" alt="image" src="https://user-images.githubusercontent.com/7405507/234263255-f650c30a-7468-4a6e-b282-f1e0d897f8b9.png">

Reply:
Added in https://github.com/elastic/kibana/pull/155375/commits/99a84e8370e549a79ce1b3b9413fe3ebd0b59d5f